src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,577 @@
+
+    // masses
+    os << "peptide_mass_tolerance = " << getDoubleOption_("precursor_mass_tolerance") << "\n";
+    os << "peptide_mass_units = " << getIntOption_("precursor_error_units") << "\n";                  // 0=amu, 1=mmu, 2=ppm
+    os << "mass_type_parent = " << getIntOption_("mass_type_parent") << "\n";                    // 0=average masses, 1=monoisotopic masses
+    os << "mass_type_fragment = " << getIntOption_("mass_type_fragment") << "\n";                  // 0=average masses, 1=monoisotopic masses
+    os << "precursor_tolerance_type = " << getIntOption_("precursor_tolerance_type") << "\n";            // 0=MH+ (default), 1=precursor m/z; only valid for amu/mmu tolerances
+    os << "isotope_error = " << getIntOption_("isotope_error") << "\n";                      // 0=off, 1=on -1/0/1/2/3 (standard C13 error), 2= -8/-4/0/4/8 (for +4/+8 labeling)
+
+    // search enzyme
+
+    // TODO: complete
+    map<String, Size> map_oms2comet;
+    map_oms2comet["Trypsin"] = 1;
+    map_oms2comet["Arg-C"] = 5;
+    map_oms2comet["Asp-N"] = 6;
+    map_oms2comet["Chymotrypsin"] = 10;
+    map_oms2comet["CNBr"] = 7;
+    map_oms2comet["Lys-C"] = 3;
+    map_oms2comet["PepsinA"] = 9;
+    map_oms2comet["Trypsin/P"] = 2;
+    map_oms2comet["no cleavage"] = 0;   
+
+    String enzyme_name = getStringOption_("enzyme");
+    Size enzyme_number = 1;
+    if (map_oms2comet.find(enzyme_name) != map_oms2comet.end())
+    {
+      enzyme_number = map_oms2comet.at(enzyme_name);
+    }
+    else
+    {
+      throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Error: Enzyme not supported. " + enzyme_name);
+    }
+
+    os << "search_enzyme_number = " << enzyme_number << "\n";                // choose from list at end of this params file
+    os << "num_enzyme_termini = " << getIntOption_("num_enzyme_termini") << "\n";                  // 1 (semi-digested), 2 (fully digested, default), 8 C-term unspecific , 9 N-term unspecific
+    os << "allowed_missed_cleavage = " << getIntOption_("allowed_missed_cleavages") << "\n";             // maximum value is 5; for enzyme search
+
+    // Up to 9 variable modifications are supported
+    // format:  <mass> <residues> <0=variable/else binary> <max_mods_per_peptide> <term_distance> <n/c-term> <required>
+    //     e.g. 79.966331 STY 0 3 -1 0 0
+    vector<String> variable_modifications_names = getStringList_("variable_modifications");
+    vector<ResidueModification> variable_modifications = getModifications_(variable_modifications_names);
+    if (variable_modifications.size() > 9)
+    {
+      throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Error: Comet only supports 9 variable modifications. " + String(variable_modifications.size()) + " provided.");
+    }
+
+    Size var_mod_index = 1;
+
+    // write out user specified modifications
+    for (; var_mod_index <= variable_modifications.size(); ++var_mod_index)
```

Comment:
ehh... vector indices are 0 based. This is certainly segfault area!
Start at `var_mod_index = 0`, Break at `var_mod_index < variable_modifications.size()`!!!!